### PR TITLE
feat(q): stub out Q LSP logic

### PR DESCRIPTION
**Pull request by rli**
This is the initial iteration of the bare minimum of process management to interact with Amazon Q logic vended by Flare.
It is a direct port of the exploratory work done in https://github.com/rli/lsp-exp/tree/master/src/main/kotlin/org/example/lsp4j

Missing are tests and any sort of edge case handling

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLanguageClientImpl.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import software.aws.toolkits.jetbrains.services.amazonq.lsp.model.aws.credentials.ConnectionMetadata
+import software.aws.toolkits.jetbrains.services.amazonq.lsp.model.aws.credentials.SsoProfileData
+import java.util.concurrent.CompletableFuture
+
+/**
+ * Concrete implementation of [AmazonQLanguageClient] to handle messages sent from server
+ */
+class AmazonQLanguageClientImpl : AmazonQLanguageClient {
+    override fun telemetryEvent(`object`: Any) {
+        println(`object`)
```
Comment (manodnyab) [inline plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLanguageClientImpl.kt:22]: nit: not needed?

File: plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLspService.kt [hunk 2]
```diff
@@ -0,0 +1,160 @@
+// Copyright 2025 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.amazonq.lsp
+
+import com.google.gson.ToNumberPolicy
```
Comment (manodnyab) [inline plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLspService.kt:6]: Should we be using gson?

File: plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLspService.kt [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            .configureGson {
+                // TODO: maybe need adapter for initialize:
+                //   https://github.com/aws/amazon-q-eclipse/blob/b9d5bdcd5c38e1dd8ad371d37ab93a16113d7d4b/plugin/src/software/aws/toolkits/eclipse/amazonq/lsp/QLspTypeAdapterFactory.java
+
+                // otherwise Gson treats all numbers as double which causes deser issues
+                it.setObjectToNumberStrategy(ToNumberPolicy.LONG_OR_DOUBLE)
+            }.traceMessages(
+                PrintWriter(
+                    object : StringWriter() {
+                        private val traceLogger = LOG.atLevel(if (isDeveloperMode()) Level.INFO else Level.DEBUG)
```
Comment (manodnyab) [inline plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLspService.kt:103]: ohh this is a good idea, maybe we can convert this to a util?

File: plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLspService.kt [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        }
+    }
+
+    companion object {
+        private val LOG = getLogger<LSPProcessListener>()
+    }
+}
+
+@Service(Service.Level.PROJECT)
+class AmazonQLspService(project: Project, private val cs: CoroutineScope) : Disposable {
```
Comment (manodnyab) [inline plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLspService.kt:73]: How would this be triggered?

File: plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLanguageClientImpl.kt [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        println(diagnostics)
+    }
+
+    override fun showMessage(messageParams: MessageParams) {
+        val type = when (messageParams.type) {
+            MessageType.Error -> NotificationType.ERROR
+            MessageType.Warning -> NotificationType.WARNING
+            MessageType.Info, MessageType.Log -> NotificationType.INFORMATION
+        }
+        println("$type: ${messageParams.message}")
```
Comment (manodnyab) [inline plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLanguageClientImpl.kt:35]: nit: remove

File: plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLanguageClientImpl.kt [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ * Concrete implementation of [AmazonQLanguageClient] to handle messages sent from server
+ */
+class AmazonQLanguageClientImpl : AmazonQLanguageClient {
+    override fun telemetryEvent(`object`: Any) {
+        println(`object`)
+    }
+
+    override fun publishDiagnostics(diagnostics: PublishDiagnosticsParams) {
+        println(diagnostics)
```
Comment (manodnyab) [inline plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLanguageClientImpl.kt:26]: nit: remove

File: plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLanguageClientImpl.kt (hunk 1 above)
Comment (rli) [reply plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLanguageClientImpl.kt:22]: we're going to keep the prints until they are implemented properly

File: plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLspService.kt (hunk 2 above)
Comment (rli) [reply plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLspService.kt:6]: yes lsp4j only works with gson

File: plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLspService.kt (hunk 4 above)
Comment (rli) [reply plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/AmazonQLspService.kt:73]: future pr
